sdl_android/src/main/java/com/smartdevicelink/encoder/VirtualDisplayEncoder.java

```diff
@@ -0,0 +1,648 @@
+    private Boolean initPassed = false;
+    private Handler uiHandler = new Handler(Looper.getMainLooper());
+    private final static int REFRESH_RATE_MS = 100;
+    private final static Object CLOSE_VID_SESSION_LOCK = new Object();
+    private final static Object START_DISP_LOCK = new Object();
+    private final static Object STREAMING_LOCK = new Object();
+
+    public class StreamingParameters {
+        protected int displayDensity = DisplayMetrics.DENSITY_HIGH;
+        protected int videoWidth = 800;
+        protected int videoHeight = 480;
+        protected int frameRate = 24;
+        protected int bitrate = 512000;
+        protected int interval = 5;
+
+        public StreamingParameters(){
+            // empty
+        }
+
+        public StreamingParameters(int displayDensity, int videoWidth, int videoHeight, int frameRate, int bitrate, int interval) {
+            this.displayDensity = displayDensity;
+            this.videoWidth = videoWidth;
+            this.videoHeight = videoHeight;
+            this.frameRate = frameRate;
+            this.bitrate = bitrate;
+            this.interval = interval;
+        }
+
+        /**
+         * Set displayDensity to a value from DisplayMetrics
+         * @param displayDensity
+         */
+        public void setDisplayDensity(int displayDensity) {
+            this.displayDensity = displayDensity;
+        }
+
+        public int getDisplayDensity() {
+            return displayDensity;
+        }
+
+        public void setVideoWidth(int videoWidth) {
+            this.videoWidth = videoWidth;
+        }
+
+        public int getVideoWidth() {
+            return videoWidth;
+        }
+
+        public void setVideoHeight(int videoHeight) {
+            this.videoHeight = videoHeight;
+        }
+
+        public int getVideoHeight() {
+            return videoHeight;
+        }
+
+        public void setFrameRate(int frameRate) {
+            this.frameRate = frameRate;
+        }
+
+        public int getFrameRate() {
+            return frameRate;
+        }
+
+        public void setBitrate(int bitrate) {
+            this.bitrate = bitrate;
+        }
+
+        public int getBitrate() {
+            return bitrate;
+        }
+
+        public void setInterval(int interval) {
+            this.interval = interval;
+        }
+
+        public int getInterval() {
+            return interval;
+        }
+    }
+
+    /**
+     * Initialization method for VirtualDisplayEncoder object. MUST be called before start() or shutdown()
+     * Will overwrite previously set videoWeight and videoHeight
+     * @param context
+     * @param videoStream
+     * @param presentationClass
+     * @param screenParams
+     * @throws Exception
+     */
+    public void init(Context context, OutputStream videoStream, Class<? extends SdlPresentation> presentationClass, ScreenParams screenParams) throws Exception {
+        if (android.os.Build.VERSION.SDK_INT < 21) {
```

Comment:
In the rest of the library we compare with the version codes, so to maintain suit this should be:

```java
if (android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.LOLLIPOP) {
```